server/src/service_manager.rs

```diff
@@ -28,22 +28,24 @@ impl ServiceManager {
         self.connection.object_server()
     }
 
-    pub fn session(&self, path: &OwnedObjectPath) -> Option<Arc<Session>> {
-        self.sessions.get(path).map(Arc::clone)
-    }
-
-    pub fn insert_session(&mut self, path: OwnedObjectPath, session: Arc<Session>) {
-        self.sessions.insert(path, session);
+    pub fn service(&self) -> Result<Arc<Service>, ServiceError> {
+        if let Some(service) = &self.service {
+            Ok(Arc::clone(service))
+        } else {
+            Err(ServiceError::ZBus(zbus::Error::FDO(Box::new(
+                zbus::fdo::Error::Failed("Service isn't initialized yet.".to_string()),
+            ))))
```

Comment:
This should be unrechable code instead of something that returns an error.

Reply:
Thanks for the prompt review. Sorry, I don't understand :)

Reply:
It is a programmer error to call any method that needs the server before the server instance is up and ready. So there is no error to report back, if that happens, you should panic using a macro called unreachable or so.

Reply:
Understood, thanks!